reader.go

```diff
@@ -100,15 +124,25 @@ func (r *Reader) ReadAll() ([][]string, error) {
 	return allRows, nil
 }
 
+// Check for BOM remove as necessary
+
 // Read reads one record from r. The record is a slice of strings with each
 // string representing one field.
 func (r *Reader) Read() ([]string, error) {
 	// TODO: Possible optimization; store the maximum number of columns for
 	// faster preallocation.
 	record := make([]string, 0, 2)
 
+	counter := 0
```

Comment:
Since this is only used to check for the first pass through the loop, consider just using a bool named something like "firstPass".

```
firstPass:= true
for {
    if firstPass {
        firstPass = false
        // code that should only be executed on the first pass
    }
    // code that should be executed on every pass
}
```

Reply:
Fixed it.